File: backend/app/api/ai.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel

from app.services.ai_service import chat, generate_flier_image, parse_flier_image

router = APIRouter()


class ChatRequest(BaseModel):
    messages: list[dict]


class FlierRequest(BaseModel):
    prompt: str


class ParseFlierRequest(BaseModel):
    image_data: str
    mime_type: str = "image/jpeg"


@router.post("/chat")
async def ai_chat(req: ChatRequest):
    reply = await chat(req.messages)
    return {"reply": reply}


@router.post("/generate-flier")
async def ai_generate_flier(req: FlierRequest):
    url = await generate_flier_image(req.prompt)
    if not url:
        raise HTTPException(status_code=503, detail="AI not configured. Set OPENAI_API_KEY.")
    return {"url": url}


@router.post("/parse-flier")
async def ai_parse_flier(req: ParseFlierRequest):
    try:
        result = await parse_flier_image(req.image_data, req.mime_type)
    except Exception:
        result = {}
    if not result:
        raise HTTPException(status_code=503, detail="AI not configured or could not read image.")
    return result

File: frontend/src/lib/api/ai.ts
import { apiClient } from "@/lib/api-client";

export async function aiChat(messages: { role: string; text: string }[]): Promise<string> {
  const formatted = messages.map((m) => ({ role: m.role, content: m.text }));
  const res = await apiClient<{ reply: string }>("/ai/chat", {
    method: "POST",
    body: { messages: formatted },
  });
  return res.reply;
}

export async function generateFlier(prompt: string): Promise<string> {
  const res = await apiClient<{ url: string }>("/ai/generate-flier", {
    method: "POST",
    body: { prompt },
  });
  return res.url;
}

export async function parseFlier(imageData: string, mimeType = "image/jpeg"): Promise<Record<string, unknown>> {
  return apiClient("/ai/parse-flier", {
    method: "POST",
    body: { image_data: imageData, mime_type: mimeType },
  });
}
